docs/src/specs/device.md

```diff
@@ -243,39 +293,86 @@ For the sake of examples, here's a few more ways of reinterpreting the logical g
 
 #### Extra Wide Grid
 ```mlir
-#tt.device<#tt.grid<1x64, (d0, d1) -> (0, d0 * 8 + d1 floordiv 8, d1 mod 8)>, [0]>
+#tt.device<
+  workerGrid = #tt.grid<1x64, (d0, d1) -> (0, d0 * 8 + d1 floordiv 8, d1 mod 8)>,
+  meshShape = 1,
+  chipIds = [0]
+>
 ```
 
 #### Extra Tall + Transposed Grid
 ```mlir
-#tt.device<#tt.grid<64x1, (d0, d1) -> (0, d1 * 8 + d0 floordiv 8, d0 mod 8)>, [0]>
+#tt.device<
+  workerGrid = #tt.grid<64x1, (d0, d1) -> (0, d1 * 8 + d0 floordiv 8, d0 mod 8)>,
+  meshShape = 1,
+  chipIds = [0]
+>
 ```
 
 #### Staircase
 ```mlir
-#tt.device<#tt.grid<8x8, (d0, d1) -> (0, d0, (d0 + d1) mod 8)>, [0]>
+#tt.device<
+  workerGrid = #tt.grid<8x8, (d0, d1) -> (0, d0, (d0 + d1) mod 8)>,
+  meshShape = 1,
+  chipIds = [0]
+>
 ```
 
 This could be an interesting starting position for data in implementing matmul as a
 systolic array in a ring topology.
 
-## Backend Lowering and Constraints
+## Lowering to TTNN
 
 While the above device attribute encoding is quite flexible, this does not
 necessarily mean the target backend can actually support all of these
-interpretations.  TTNN backend will be relatively constrained to support only
-the specialized grid topologies that are supported by the API.
-
-### TTNN
-
-TODO:
-
-- Multi-device
-- Grid orientation
-- Height / Width sharded
-- TTNN Generic
-
-### TTMetal
+interpretations.  TTNN backend will be constrained to support only the
+specialized grid topologies that are supported by the API.
+
+### Grid/Shard Orientation
+
+TODO
+
+### Multi-device
+
+Please refer to [TTNN Mesh Programming Docs](https://github.com/tenstorrent/tt-metal/blob/main/tech_reports/Programming%20Mesh%20of%20Devices/Programming%20Mesh%20of%20Devices%20with%20TT-NN.md)
+for more information on how to program multi-device systems with TTNN API.
+
+Multi-device TTNN dialect will try and stay as close to the TTNN API as
+possible.  Let's consider what this looks like from the compiler and runtime
+perspectives:
+
+#### Compiler
+
+- **Device Creation**: The TTNN device in the compiler is exactly the same attribute
+  from the ttir dialect.  It will encode the `meshShape` into the flatbuffer
+  which can be directly used to program `::ttnn::MeshShape`.
+- **Tensor Layout**: Again, the tensor layout is inherited in TTNN dialect from the
+  ttir dialect.  The grid attribute in the tensor layout can be trivially
+  divided by `meshShape` to determine the shape of the tensor slice on each device.
+  Broadcasting rules can be applied to determine which [Distribution Strategy](https://github.com/tenstorrent/tt-metal/blob/main/tech_reports/Programming%20Mesh%20of%20Devices/Programming%20Mesh%20of%20Devices%20with%20TT-NN.md#3-distributing-tensor-to-meshdevice)
+  to use:
+    - **Sharding**: If the tensor grid is > 1 along the `meshShape` dimensions,
```

Comment:
Can you use Split or some other naming, Sharding might be overloaded term here...